route/v1/types.go

```diff
@@ -44,6 +44,7 @@ type Route struct {
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
 	// spec is the desired state of the route
+	// +kubebuilder:validation:XValidation:rule="!has(self.tls) || self.tls.termination != 'passthrough' || (size(self.httpHeaders.actions.response) == 0 || size(self.httpHeaders.actions.request) == 0)",message="header actions are not permitted when tls termination is passthrough."
```

Comment:
I think you need to update the godoc on the termination field to mention that this cannot be used with `httpHeader.actions`

Also, is this validation correct, I think we need the headers to both be 0 length right? But currently you have an `||`? 
```suggestion
	// +kubebuilder:validation:XValidation:rule="!has(self.tls) || self.tls.termination != 'passthrough' || (size(self.httpHeaders.actions.response) == 0 && size(self.httpHeaders.actions.request) == 0)",message="header actions are not permitted when tls termination is passthrough."
```